documentation/troubleshooting.md

```diff
@@ -0,0 +1,90 @@
+title: Troubleshooting Kubernetes vSphere Cloud Provider
+---
+This document is aimed to help debug issues in the Kuberentes vSphere Cloud Provider.
+
+For general Kubernetes Issue, Please visit [https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/)
+
+## What logs can I inspect?
+For digging deeper and help gather data for the technical support we need to see the log files.
+
+For vSphere Cloud Provider most often we need to look into Controller-Manager, Kubelet and API-server logs.
+
+Logs for Controller-Manager and API server should be available on the Master node VM and Logs for Kubelet is located on both Master and Worker Nodes.
+
+To locate the log files for Controller-Manager and API server, we need to find the `Container ID` from Pods. `Container ID` can be obtained by inspecting Pod running in the kube-system namespace or using `docer ps -a` command
+
+Login into the master node and execute following command.
+
+```
+# kubectl describe pod kube-controller-manager-kubeadm-master --namespace=kube-system | grep "Container ID"
+    Container ID:  docker://74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a
+```
+
+if you are unable to find controller-manager pod, use `docker ps -a` as below to grab `Container ID`.
+
+```
+# docker ps | grep controller-manager
+74a15e75d136        gcr.io/google_containers/kube-controller-manager-amd64   "kube-controller-m..."   23 minutes ago      Up 23 minutes                           k8s_kube-controller-manager_kube-controller-manager-kubeadm-master_kube-system_22d908089b353bf8749a89843022bff3_0
+12df71e6d4a8        gcr.io/google_containers/pause-amd64:3.0                 "/pause"                 23 minutes ago      Up 23 minutes                           k8s_POD_kube-controller-manager-kubeadm-master_kube-system_22d908089b353bf8749a89843022bff3_0
+
+
+
+# docker inspect 74a15e75d136 | grep Id
+        "Id": "74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a",
+```
+Once the `Container ID` is obtained, logs can be obtained from `/var/lib/docker/containers/` directory. Docker creates the directory for each container and keeps logs in that directory.
+
+```
+# ls /var/lib/docker/containers/74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a
+74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a-json.log  checkpoints  config.v2.json  hostconfig.json
+```
+
+**Note:** On `systemd-based` setup, you may need to use `journalctl` instead. You need to know the service name to get the logs for controller-manager.
+- Here is the example.
+  ```
+  journalctl -u kube-controller-manager.service > controller-manager.log
+  ```
+
+Kubelet generally runs as the service and not in the container or Pod in the Kubernetes cluster. Kubelet logs can be obtained using `journalctl` as shown below.
+
+```
+journalctl -u kubelet > kubelet.log
+```
+
+## How can I modify the log level?
```

Comment:
Let's not use "I"/ "We" in technical documentation.

Rewording suggestion : Modifying log level

Reply:
Changed to `Modify log level`